lib/gds_api/test_helpers/email_alert_api.rb

```diff
@@ -68,6 +68,54 @@ def stub_email_alert_api_has_subscription(
           .to_return(status: 200, body: response)
       end
 
+      # Stubs the API responses as if each subscription happened in the order they are passed.
+      # Useful if you need to query the '/latest' endpoint of a subscription.
+      # Takes an array of hashes.
+      #
+      # @example
+      #  stub_email_alert_api_has_subscriptions([
+      #    {
+      #      id: 'id-of-my-subscriber-list',
+      #      frequency: 'weekly',
+      #      ended: true,
+      #    },
+      #    {
+      #      id: 'id-of-my-subscriber-list',
+      #      frequency: 'daily',
+      #    },
+      #  ])
+      #
+      # @param subscriptions [Array]
+      def stub_email_alert_api_has_subscriptions(subscriptions)
+        subscriptions.map! { |subscription| apply_subscription_defaults(subscription) }
+        subscriptions.each do |id, params|
+          latest_id, latest_params = get_latest_matching(params, subscriptions)
+          stub_request(:get, "#{EMAIL_ALERT_API_ENDPOINT}/subscriptions/#{id}")
+            .to_return(status: 200, body: get_subscription_response(id, params).to_json)
+          stub_request(:get, "#{EMAIL_ALERT_API_ENDPOINT}/subscriptions/#{id}/latest")
+            .to_return(status: 200, body: get_subscription_response(latest_id, latest_params).to_json)
+        end
+      end
+
+      def apply_subscription_defaults(subscription)
+        parameters = {
+          title: "Some title",
+          subscriber_id: 1,
+          subscriber_list_id: 1000,
+          ended: true,
+        }.merge(subscription)
+        id = parameters.delete(:id) # Otherwise `ArgumentError: unknown keyword: id`
```

Comment:
I'm not clear on why the `id` value has to be deleted from `parameters`.  Where does the `ArgumentError` occur if this isn't deleted?

Reply:
Fair point - I'll make the comment more descriptive 👍